### DISCUSSION TITLE
question about IEA 15MW simulation using Simulink: 'S-Function' FASTSFunc 'reports an error

### DISCUSSION THREAD

**Discussion by yangyangyang98** [OpenFAST/openfast#2506] [Q&A]
Dear Jonkman@jjonkman
When I was running the IEA 15MW simulation using Simulink, I encountered the following issue: 'S-Function' FASTSFunc 'reports an error:
FAST_InitializeAll:ED_Init:ED_ReadInput:ReadPrimaryFile:OpenFInpFile:Cannot open file ".\IEA-15-240-RWT-UMaineSemi_ElastoDyn.dat"。 But I am certain that the ElastoDyn module is located in the current folder.

I am using IEA-15-240-RWT-1.1.8 version, and openfast-x64.exe and FASTSFunc.mexw64 are v3.5.1 versions. When I run it with cmd, it can run normally, but the above problem occurs in Simulink. I suspect it is a problem with FASTS_SFunc.mexw64, but I am not sure yet

**Comment by jjonkman** [comment]
Dear @yangyangyang98,

Can you clarify how you are running OpenFAST within Simulink (e.g. using _OpenLoop.mdl_ and _Run_OpenLoop.m_) and where the files (_.mdl_, _.m_, _.mexw64_, and _.dll_) are relative to your OpenFAST input files (_.fst_ and _.dat_)?

Best regards

**Comment by yangyangyang98** [comment]
Dear @jjonkman
Thank you for your prompt response. My issue has been resolved, and the reason for the error was that the file name in the folder was too long.

Besides，I have a new question. I would like to inquire about the platform-pitch natural frequency of the IEA 22MW semi-submersible floating wind turbine, as well as the natural frequency and damping ratio used by the pitch controller，as I could not find them in “Definition of the IEA Wind 22-Megawatt Offshore Reference Wind Turbine”. Thanks.

**Reply by jjonkman** [reply]
Dear @yangyangyang98,

I was not the one who developed the IEA Wind 22-MW RWT or its semi-submersible.  I would suggest asking questions about the design in the IEA Wind 22-MW RWT repository: https://github.com/IEAWindTask37/IEA-22-280-RWT/issues.

Best regards,

**Reply by YongxiangLei** [reply]
Dear @yangyangyang98 
I have a similar issue, did you find a solution?

I would also like to use Matlab S-Function for 22MW wind turbine. Is it possible to use S-Function to get OpenFast running results?

Best regards,
Yong

**Reply by jjonkman** [reply]
Dear @YongxiangLei,

The OpenFAST interface to Simulink/MATLAB is generic and not turbine-dependent.  What issue are you running into?

Best regards,
